modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -73,12 +81,59 @@ public AssetManagerWithSecurity(
     this.secSvc = secSvc;
   }
 
-  @Override public Snapshot takeSnapshot(String owner, MediaPackage mp) {
-    final AccessControlList acl = authSvc.getActiveAcl(mp).getA();
+  private boolean isAuthorizedByAcl(AccessControlList acl, String action) {
     final User user = secSvc.getUser();
     final Organization org = secSvc.getOrganization();
-    if (AccessControlUtil.isAuthorized(acl, user, org, WRITE_ACTION)) {
+    return AccessControlUtil.isAuthorized(acl, user, org, action);
+  }
+
+  private boolean isAuthorizedByAcl(MediaPackage mp, String action) {
+    final AccessControlList acl = authSvc.getActiveAcl(mp).getA();
+    return isAuthorizedByAcl(acl, action);
+  }
+
+  private boolean isAuthorizedByAcl(Version version, String mpId, String action) {
+    final AQueryBuilder q = q();
+    final AResult r = q.select(q.snapshot())
+            .where(q.mediaPackageId(mpId))
+            .run();
+    if (r.getSize() == 1) {
+      final Opt<Snapshot> snapShot = r.getRecords().head2().getSnapshot();
+      if (snapShot.isSome()) {
+        final MediaPackage mp = snapShot.get().getMediaPackage();
+        String xacmlId = null;
+        for (MediaPackageElement mpe : mp.getElements()) {
+          if (mpe.toString().contains("security-policy-episode")) {
+            xacmlId = mpe.getIdentifier();
+            break;
+          }
+        }
+        if (xacmlId != null) {
+          Opt<Asset> secAsset = super.getAsset(version, mpId, xacmlId);
+          if (secAsset.isSome()) {
+            InputStream in = secAsset.get().getInputStream();
+            final AccessControlList acl = authSvc.getAclFromInputStream(in).getA();
+            return isAuthorizedByAcl(acl, action);
+          }
+          return false;
```

Comment:
Please try to write the return statment directly behind the closed if clause. 
I don't know how to do it the best way. My be by reversing the if condition?

Reply:
Could just use a single "return false;" at the end rather than all these following. Most of the else conditions aren't required then.